File: app/models/adoption_model.py
from sqlalchemy import Column, Integer, ForeignKey, DateTime, Boolean
from sqlalchemy.orm import relationship
from datetime import datetime
from app.db.base import Base

class Adoption(Base):
    __tablename__ = "adoptions"

    id = Column(Integer, primary_key=True)

    pet_id = Column(Integer, ForeignKey("pets.id"), nullable=False)
    adoptante_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    status_id = Column(Integer, ForeignKey("adoption_status_model.id"), nullable=False)

    fecha_solicitud = Column(DateTime, default=datetime.utcnow)
    fecha_respuesta = Column(DateTime)

    cedula_url = Column(String)
    recibo_url = Column(String)

    quiere_tracker = Column(Boolean, default=False)
    deleted_at = Column(DateTime, nullable=True)

    pet = relationship("Pet", back_populates="adoptions")
    status = relationship("AdoptionStatus")
    adoptante = relationship("User")

File: FrontendV2/src/app/shared/models/adoption.model.ts
export interface AdoptionStatus {
  id: number;
  name: 'PENDING' | 'APPROVED' | 'REJECTED';
}

export interface AdoptionUser {
  id: number;
  name: string;
  last_name: string;
  email: string;
}

export interface Adoption {
  // Campos obligatorios
  id: number;
  pet_id: number;
  adoptante_id: number;
  status_id: number;
  fecha_solicitud: string;

  // Campos opcionales
  fecha_respuesta?: string | null;
  cedula_url?: string | null;
  recibo_url?: string | null;
  deleted_at?: string | null;

  // Relaciones (vienen del backend como objetos anidados)
  pet?: { id: number; name: string; image_url?: string | null };
  status?: AdoptionStatus;
  adoptante?: AdoptionUser;
}
